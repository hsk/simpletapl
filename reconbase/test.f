/* Examples for testing */

 lambda x:A. x;

lambda x:Bool. x;
(lambda x:Bool->Bool. if x false then true else false) 
  (lambda x:Bool. if x then false else true); 

lambda x:Nat. succ x;
(lambda x:Nat. succ (succ x)) (succ 0); 
