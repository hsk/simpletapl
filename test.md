# ここのところやってたこと

- PrologにTAPLのOCamlコードを移植。
- 移植したコードを更に短くした。
- 本質的な部分の取り出しを行った。
