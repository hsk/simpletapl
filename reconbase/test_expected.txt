lambda x:A.x : A -> A
lambda x:Bool.x : Bool -> Bool
true : Bool
lambda x:Nat.succ x : Nat -> Nat
3 : Nat
